Dev

fedossa opened the pull request:
Last changes from me prior to 0.2.0

fedossa committed:
Bug: botex.sqlite used instead of BOTEX_DB

fedossa committed:
Refactor - removed double import of botex

fedossa committed:
Update dependencies: Added pydantic to main, otree and mkdocs-click to dev dependency.

fedossa committed:
Update llama.cpp parameter retrieval, this restricts the llama.cpp version to b4285 or later.

fedossa committed:
Remove the redundant env variable setting

fedossa committed:
Relink GGUFParser, context length will once more be read from the model if not provided.

fedossa committed:
Small bug in llamacpp test

fedossa committed:
Added info on set_parms_from_running_api key error.

joachim-gassen committed:
Rewrote code to also work with older llama.cpp versions (I hope)

joachim-gassen reviewed (commented):
Pllease a take quick look at my commit. I tried to rewrite the llama.cpp code so that it should still work with an old llama.cpp version. Can you verify whether it also works with the newer versions?

fedossa committed:
Included an unknown model catch.

At one point the model_path was completely removed from the llama.cpp endpoint, so included an unknown model catch to deal with this and stop botex from erroring out.

fedossa commented:
I have looked over the commit and have extended it, to cover the case when the model_path was completely removed from llama.cpp server endpoint. I have also rebased.
